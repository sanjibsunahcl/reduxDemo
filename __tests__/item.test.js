import React from 'react';
import {create, act} from 'react-test-renderer';
import Items from '../src/screen/items';

// jest.runAllTimers();
//Snap shot test
it('Item snapshot', () => {
  const tree = create(<Items />);
  expect(tree).toMatchSnapshot();
});

test('button press', () => {
  //button press
  const tree = create(<Items />);
  const button = tree.root.findByProps({testID: 'myButton'}).props;
  act(() => button.onPress());
  const text = tree.root.findByProps({testID: 'myText'}).props;
  expect(text.children).toEqual('button pressed');
});

test('call timeout', () => {
  //to perform the asynchronous like useffect timeout etc.
  const tree = create(<Items />);

  act(() => jest.runAllTimers());
  const text = tree.root.findByProps({testID: 'timeOutText'}).props;
  // console.log('hfhjfdhjhjf' + JSON.stringify(text));
  expect(text.children).toEqual('timeout is called');
});

describe('Check count', () => {
  test('use effect set up', () => {
    const tree = create(<Items />);
    act(() => jest.runAllTimers());
    const text = tree.root.findByProps({testID: 'dataId'}).props;
    // console.log(text.children);
    expect(text.children).toEqual(20);
  });
  test('button press', () => {
    const tree = create(<Items />);
    const button = tree.root.findByProps({testID: 'countButton'}).props;
    act(() => button.onPress());
    const text = tree.root.findByProps({testID: 'countText'}).props;
    expect(text.children).toEqual(5);
  });
});
