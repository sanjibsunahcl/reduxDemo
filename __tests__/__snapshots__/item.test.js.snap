// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    testID="dataId"
  >
    
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 18,
      }
    }
  >
    Items
  </Text>
  <Text
    testID="timeOutText"
  >
    
  </Text>
  <Text
    testID="myText"
  >
    
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Click
      </Text>
    </View>
  </View>
  <Text
    testID="countText"
  >
    4
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="countButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press
      </Text>
    </View>
  </View>
</View>
`;
